controllers/account.go

```diff
@@ -127,32 +132,31 @@ func (c *ApiController) AuthGithub() {
 		c.Data["json"] = resp
 		c.ServeJSON()
 		return
-	} else {
-		if addition == "signup" {
-			Github := object.HasGithub(tempUserAccount.Login)
-			if Github != "" {
-				c.SetSessionUser(Github)
-				util.LogInfo(c.Ctx, "API: [%s] signed in", Github)
-				res.IsSignedUp = true
-			} else {
-				c.SetSessionUser(Github)
-				util.LogInfo(c.Ctx, "API: [%s] signed in", Github)
-				res.IsSignedUp = true
-			}
-			res.Addition = tempUserAccount.Login
-			res.Avatar = tempUserAccount.AvatarUrl
-			if Github == "438561537" || Github == "hsluoyz" || Github == "nodece" || Github == "BetaCat0" {
-				res.IsAdmin = true
-			} else {
-				res.IsAdmin = false
-			}
-			_ = object.LinkUserAccount(res.Addition, res.Email, res.Avatar, res.IsAdmin)
-			resp = Response{Status: "ok", Msg: "success", Data: res}
+	}
+	if addition == "signup" {
+		Github := object.HasGithub(tempUserAccount.Login)
+		if Github != "" {
+			c.SetSessionUser(Github)
+			util.LogInfo(c.Ctx, "API: [%s] signed in", Github)
+			res.IsSignedUp = true
+		} else {
+			c.SetSessionUser(Github)
+			util.LogInfo(c.Ctx, "API: [%s] signed in", Github)
+			res.IsSignedUp = true
 		}
-		c.Data["json"] = resp
-		c.ServeJSON()
-		return
+		res.Addition = tempUserAccount.Login
+		res.Avatar = tempUserAccount.AvatarUrl
+		if Github == "438561537" || Github == "hsluoyz" || Github == "nodece" || Github == "BetaCat0" {
```

Comment:
What's this? Hard-coding admin user names into code is not a good practice..

Reply:
Yes, it needs a more flexible role management method...... 
I may do more future jobs to improve this project.

Reply:
I agree with @hsluoyz , I think you can think of using a config file instead of hard-coding temporarily(this allow you writing a simple auth module for admin management )